lib/src/boundimage.rs

```diff
@@ -0,0 +1,171 @@
+use crate::task::Task;
+use anyhow::{Context, Result};
+use camino::Utf8Path;
+use fn_error_context::context;
+use ostree_ext::ostree::Deployment;
+use ostree_ext::sysroot::SysrootLock;
+use regex::Regex;
+use rustix::fs::{OFlags, ResolveFlags};
+use std::fs;
+use std::fs::File;
+use std::io::Read;
+use std::os::unix::io::AsFd;
+use std::path::Path;
+
+const BOUND_IMAGE_DIR: &'static str = "usr/lib/bootc/bound-images.d";
```

Comment:
We discussed making this something like `usr/lib/bootc-experimental/bound-images.d` so we can allow experimenting with it without committing immediately to the interface.